### PR TITLE
feat: list all addresses in vault

### LINKED ISSUE #413: List all the addresses used in a vault
Follow up of #410

### PR DESCRIPTION
**Pull request by djaiss**
Close #413 

This PR
- rewrites the Address module using `setup script` of Vue 3
- adds a new Report tab that shows all the countries and cities used in the vault
- grabs information from Wikipedia when viewing a country or a city
- changes the behaviour of deleting an address: if no other contact uses an address, we delete the address entirely.

### CODE REVIEW

File: app/Domains/Vault/ManageReports/Web/ViewHelpers/ReportAddressIndexViewHelper.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            ->distinct('city')
+            ->get()
+            ->map(fn ($address) => [
+                'id' => $address->id,
+                'name' => Str::ucfirst($address->city),
+                'contacts' => $address->contacts_count,
+                'url' => [
+                    'index' => route('vault.reports.addresses.cities.show', [
+                        'vault' => $vault->id,
+                        'city' => urlencode(utf8_encode($address->city)),
```
Comment (asbiin) [inline app/Domains/Vault/ManageReports/Web/ViewHelpers/ReportAddressIndexViewHelper.php:29]: I don't think `urlencode` is necessary with `route` method.

File: app/Helpers/WikipediaHelper.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        $query = http_build_query([
+            'action' => 'query',
+            'prop' => 'description|pageimages',
+            'titles' => $topic,
+            'pithumbsize' => 400,
+            'format' => 'json',
+        ]);
+
+        $url = 'https://en.wikipedia.org/w/api.php?'.$query;
```
Comment (asbiin) [inline app/Helpers/WikipediaHelper.php:28]: I don't think it's a good idea to hardcode the english wikipedia url.

File: app/Helpers/WikipediaHelper.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+        $response = Http::get($url);
+        $response->throw();
```
Comment (asbiin) [inline app/Helpers/WikipediaHelper.php]: you can do
```suggestion
        $response = Http::get($url)->throw();
```

File: resources/js/Shared/Modules/Addresses.vue [hunk 4]
```diff
@@ -1,3 +1,142 @@
...
+      editedAddressId.value = 0;
+    })
+    .catch((error) => {
+      loadingState.value = '';
+      form.errors = error.response.data;
+    });
+};
+
+const destroy = (address) => {
+  if (confirm('Are you sure? This can not be undone.')) {
```
Comment (asbiin) [inline resources/js/Shared/Modules/Addresses.vue]: Can you use the `ConfirmationModal` module instead?

**Review by asbiin — CHANGES_REQUESTED** [review]
See my comments

File: app/Domains/Vault/ManageReports/Web/ViewHelpers/ReportAddressIndexViewHelper.php (hunk 1 above)
Comment (djaiss) [reply app/Domains/Vault/ManageReports/Web/ViewHelpers/ReportAddressIndexViewHelper.php:29]: It is unfortunately.

File: resources/js/Shared/Modules/Addresses.vue (hunk 4 above)
Comment (djaiss) [reply resources/js/Shared/Modules/Addresses.vue]: Done

File: app/Helpers/WikipediaHelper.php (hunk 2 above)
Comment (djaiss) [reply app/Helpers/WikipediaHelper.php:28]: I agree. Although I'll do it in #425

**Pull request merged**